src/binaryapi/TradingTimes.js

```diff
@@ -0,0 +1,163 @@
+import EventEmitter from 'event-emitter-es6';
+
+class TradingTimes {
+    static get EVENT_MARKET_OPEN_CLOSE_CHANGE() { return 'EVENT_MARKET_OPEN_CLOSE_CHANGE'; }
+    static get FEED_UNAVAILABLE() { return 'chartonly'; }
+    lastUpdateDate = new Date().toISOString().substring(0, 10);
+
+    constructor(api) {
+        this._api = api;
+        this._emitter = new EventEmitter({ emitDelay: 0 });
+    }
+
+    destructor() {
+        if (this._updateTimer) {
+            clearTimeout(this._updateTimer);
+        }
+    }
+
+    async initialize() {
+        if (!this._tradingTimesMap) {
+            await this._updateTradeTimes();
+
+            const periodicUpdate = async () => {
+                const changed = this._updateMarketOpenClosed();
+                if (Object.keys(changed).length > 0) {
+                    this._emitter.emit(TradingTimes.EVENT_MARKET_OPEN_CLOSE_CHANGE, changed);
+                }
+                let nextUpdate = this._nextUpdateDate();
+
+                if (!nextUpdate) {
+                    const now = new Date();
```

Comment:
@brucebinary you could move this to `_nextUpdateDate`,  it's easier to read, if the `nextdate` was set inside th `_nextUpdateDate`

Reply:
@MahboobehMohammadi good point... 

Putting it inside `_nextUpdateDate` causes the unit tests to fail though. Not quite sure what went wrong there. Some race condition thing happening. I'll take a look again tmrw.